maps/src/main/res/layout/offline_map_layers_importer.xml

```diff
@@ -24,118 +24,122 @@
         <include layout="@layout/app_bar_shadow"/>
     </com.google.android.material.appbar.AppBarLayout>
 
-    <ScrollView
-        android:layout_width="match_parent"
-        android:layout_height="0dp"
-        app:layout_constraintBottom_toBottomOf="@+id/shadow_up"
-        app:layout_constraintEnd_toEndOf="parent"
+    <com.google.android.material.textview.MaterialTextView
+        android:id="@+id/layers_title"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:text="@string/layers_list_title"
+        android:textAppearance="?textAppearanceTitleLarge"
+        android:layout_marginHorizontal="@dimen/margin_standard"
+        android:layout_marginTop="@dimen/margin_small"
+        app:layout_constraintTop_toBottomOf="@id/toolbar_container"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/add_project_toolbar">
+        app:layout_constraintEnd_toEndOf="parent" />
 
-        <androidx.constraintlayout.widget.ConstraintLayout
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:id="@+id/layers_container"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        app:layout_constrainedHeight="true"
+        app:layout_constraintVertical_bias="0"
+        app:layout_constraintStart_toStartOf="@id/layers_title"
+        app:layout_constraintEnd_toEndOf="@id/layers_title"
+        app:layout_constraintBottom_toTopOf="@id/container"
+        app:layout_constraintTop_toBottomOf="@id/layers_title"
+        app:layout_constraintVertical_chainStyle="packed">
+
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/layers"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/margin_small"
+            app:layout_constrainedHeight="true"
+            app:layout_constraintVertical_bias="0"
+            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <com.google.android.material.progressindicator.CircularProgressIndicator
+            android:id="@+id/progress_indicator"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginVertical="@dimen/margin_small"
+            app:layout_constrainedHeight="true"
+            app:layout_constraintVertical_bias="0"
+            android:indeterminate="true"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
I think this (the "Select layers access" section) should be anchored to the top of the "Cancel"/"Add layers" buttons so that it's in a consistent place regardless of how large the list is.

cc/ @alyblenkin as you may disagree?

Reply:
If the list of layers to import is short it's just below the list but if it's very long then this section is not pushed off the screen and is anchored to the top of the "Cancel"/"Add layers" buttons. Did you test it and it didn't work like that?

Reply:
Yeah, I see that's the current behaviour from the layout. I think we should always anchor it to the bottom buttons though. In general, I think keeping controls in a consistent place if we can is better as it helps users who have carried the action out multiple times.

Reply:
Usually, people won't add many layers at once I guess. It will be one maybe two. In such cases, it doesn't look good if there is a list of layers to import (short list) and a lot of empty space, and that section on top of the bottom menu. But that's my opinion.

Reply:
> Usually, people won't add many layers at once I guess. It will be one maybe two. In such cases, it doesn't look good if there is a list of layers to import (short list) and a lot of empty space, and that section on top of the bottom menu. But that's my opinion.

Agreed. Our assumption is that people will only add a few at a time. 

Just to clarify, are we talking about "proposal" below? If our assumption is they will only add a few at a time, then in most cases, there will be a lot of empty space. So by anchoring "Layer access section" to the top of the buttons are we would be optimizing for the worst case scenario with 5-10 layers? Could we make the top section scrollable if there is a long list of 5+ layers so the controls are always visible? 


<img width="460" alt="Screenshot 2024-06-14 at 9 18 48 AM" src="https://github.com/getodk/collect/assets/7871340/11aa9757-2be7-4926-a68b-fc5211d23e1a">

Reply:
> Just to clarify, are we talking about "proposal" below? If our assumption is they will only add a few at a time, then in most cases, there will be a lot of empty space. So by anchoring "Layer access section" to the top of the buttons are we would be optimizing for the worst case scenario with 5-10 layers?

Correct.

> Could we make the top section scrollable if there is a long list of 5+ layers so the controls are always visible?

It works like that now. If the list is short it looks like above (the first screenshot) and if it's long it takes as much space as possible without pushing the section below off the screen.

So above I said that I like the way I've implemented it and we wanted to confirm you like it too. It looks like you agree.

Reply:
We can always iterate and anchor it to the bottom if we find people are adding many layers at once! I honestly think both work fine, but I'm leaning towards the first option because it looks like something might be broken or missing when you only have a few layers.